### ISSUE TITLE
Feature: Filter files using .gitignore to exclude superfluous files

### ISSUE THREAD

**Issue by alex-ip** [Language-Research-Technology/crate-o#82]
Git repositories typically use a .gitignore files to exclude files from commits.

Crate-O currently adds **all** files when "Load Files" is selected, so it would be very useful to apply the same exclusion rules if possible.

A typical .gitignore file would be as follows:
`# Byte-compiled / optimized / DLL files
__pycache__/
*.py[cod]
*$py.class
dataset_metadata_cache.sqlite

# C extensions
*.so

# Distribution / packaging
.Python
env/
build/
develop-eggs/
dist/
downloads/
eggs/
.eggs/
lib/
lib64/
parts/
sdist/
var/
*.egg-info/
.installed.cfg
*.egg

# PyInstaller
#  Usually these files are written by a python script from a template
#  before PyInstaller builds the exe, so as to inject date/other infos into it.
*.manifest
*.spec

# Installer logs
pip-log.txt
pip-delete-this-directory.txt

# Unit test / coverage reports
htmlcov/
.tox/
.coverage
.coverage.*
.cache
nosetests.xml
coverage.xml
*,cover

# Translations
*.mo
*.pot

# Django stuff:
*.log

# Sphinx documentation
docs/_build/

# PyBuilder
target/

# pycharm project files
.idea/

# iPython checkpoints
*-checkpoint.ipynb

# Temporary files
/Temp/

# Tiling tests
examples/dataset_lists
examples/point_lists
examples/tiles
`

**Comment by r-tae** [comment]
I think we should use a `.crateoignore` file instead. There may be cases where you want git to ignore files that Crate-O should read, using a separate Crate-O-specific ignore file is better than making that impossible

**Comment by alex-ip** [comment]
G'day River - what specific examples did you have in mind? Things like sensitive configuration files? 
Do you have a specific example where lots of files would be mistakenly excluded from the catalogue because of .gitignore?

I am inclined to disagree with the .crateoignore idea and stick with .gitignore for the following reasons:
1.  I don't think we want to introduce a new, custom configuration file to the repo which will need to be created and kept consistent with other files (including with .gitignore).
2. We should never want to catalogue (sometimes numerous) files which are a by-product of a build or test which would normally be specified in .gitignore.
4. Any individual files omitted from the "File Load" operation because of .gitignore (like sensitive configuration files) can be easily added in the UI, and once there, they are in the crate for good. It's easier to do that than manually delete a large number of rubbish files like we currently have to.
5. When the RO-crate is cataloguing files in a git repository, it doesn't really make a lot of sense to catalogue local files which will never appear in the remote repository.

We could look at providing an option to ignore the .gitignore file, but it might not be worth the bother - the file catalogue may need to be tweaked manually in either case, but I'm aiming for maximum gain with minimum pain. 

That's my 2c worth, anyway. I think it's certainly worth discussing options before implementing anything, though.

**Comment by moisbo** [comment]
Also do you think this would be with user input or something in the profile?

We dont have a file picker also. Maybe something that we can do?

**Comment by r-tae** [comment]
I often have a gitignored `data/` directory in my repos

You're probably mainly thinking of using Crate-O to catalogue software source code, but the folders I'm dealing are repos for other reasons and I wouldn't want the gitignore to be used by crate-o just because it exists.

There is plenty of prior art: `.dockerignore`, `.eslintignore`, `.prettierignore`, `.railwayignore`, `.npmignore` etc

**Comment by alex-ip** [comment]
Fair enough - are they all the same format?

**Comment by r-tae** [comment]
Yeah that could work, ignore files are almost always in the same format
